drivers/stepper/adi_tmc/adi_tmc50xx_stepper_controller.c

```diff
@@ -65,7 +83,8 @@ static int tmc50xx_write(const struct device *dev, const uint8_t reg_addr, const
 
 	k_sem_take(&data->sem, K_FOREVER);
 
-	err = tmc_spi_write_register(&bus, TMC5XXX_WRITE_BIT, reg_addr, reg_val);
+	err = tmc_spi_write_register(&bus, TMC5XXX_WRITE_BIT, reg_addr, reg_val,
+				     parse_tmc50xx_spi_status);
```

Comment:
I would say, let the spi block return the status bytes, rather than this callback mechanism. It would be much more simpler. On the driver layer, one can parse out the bits, 0, 1 and 2(in the case of tmc50xx series).